### DISCUSSION TITLE
Playing Randomly From MPD Music Database

### DISCUSSION THREAD

**Discussion by whellm** [MusicPlayerDaemon/MPD#1666] [General]
Hello - 

Is there an MPD client that allows playing music randomly from the mpd database?  My instance of MPD runs "headless" from my music server.  I'd like the ability to set mpd to just play random selections continuously from the music database, and be able to start and stop this process using a remote client on my laptop of tablet.  

Thanks for any insight!

**Comment by martadinata666** [comment]
in my setup, i load all songs in my library, and set mpd playback mode to repeat and random. so the client itself pretty irrelevant I assume.

**Reply by whellm** [reply]
I did try that, but received an error - using the Cantata client - that the playlist is too large.

**Comment by Emilot** [comment]
https://github.com/jcorporation/mympd

**Reply by whellm** [reply]
Mympd looks interesting but the install instructions look a bit over my head.   Any other thought on this?

**Reply by martadinata666** [reply]
ncmpcpp? a nurses gui, but afaik this is most reliable I have been using

**Reply by Emilot** [reply]
> Mympd looks interesting but the install instructions look a bit over my head. Any other thought on this?

If you have a Linux PC it is rather simple. Download the git repo and run the installation script "./build.sh releaseinstall"

What distro do you use?

**Reply by whellm** [reply]
Hi again - I run MX Linux 21, based on Debian.  I tried to follow the instructions for pre-built packages and it failed.

**Reply by jcorporation** [reply]
You can get help installing myMPD here: https://github.com/jcorporation/myMPD/discussions

**Comment by critkitten** [comment]
You can use mpc and bash like:

**mpc listall | shuf | head -n 10**

To list 10 Random tracks depend on shuf.

add to pipe **sed 's![^/]*$!!'** to shrink backwards until first /

now you can add in same pipe **mpc add** and have full albums.

**Reply by whellm** [reply]
I'm running MPD on a headless server, so I suspect entering commands as shown above is not an option for me.  The functionality would ideally be contained in an mpd client.
